lib/sdk/remote/utils.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const { Class } = require('../core/heritage');
+const { List, addListItem, removeListItem } = require('../util/list');
+const { emit } = require('../event/core');
+
+function forwardEvents(source, target, extraArg) {
+  let listener = (event, ...args) => {
+    emit(target, event, extraArg, ...args);
+  }
+
+  source.on('*', listener);
+
+  return function detach() {
+    source.off('*', listener);
+  }
+}
```

Comment:
I think you may be able to just modify `require("sdk/event/utils").pipe` to support this use case.